File: client/java/src/main/java/com/be/mesh/client/struct/Page.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.struct;

import com.be.mesh.client.annotate.Index;
import lombok.Data;

import java.io.Serializable;

/**
 * @author coyzeng@gmail.com
 */
@Data
public class Page<T> implements Serializable {

    private static final long serialVersionUID = -2185748030617065721L;
    @Index(0)
    private String sid;
    @Index(5)
    private long index;
    @Index(10)
    private long limit;
    @Index(15)
    private long total;
    @Index(20)
    private boolean next;
    @Index(25)
    private T data;

}

File: client/python/mesh/kinds/page.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from typing import Generic

from mesh.macro import T, index as idx, serializable


@serializable
class Page(Generic[T]):

    @idx(0)
    def sid(self) -> str:
        return ''

    @idx(5)
    def index(self) -> int:
        return 0

    @idx(10)
    def limit(self) -> int:
        return 0

    @idx(15)
    def total(self) -> int:
        return 0

    @idx(20)
    def next(self) -> bool:
        return False

    @idx(25)
    def data(self) -> T:
        return None
